smee/internal/dhcp/handler/reservation/option.go

```diff
@@ -123,3 +130,46 @@ func (h *Handler) bootfileAndNextServer(ctx context.Context, pkt *dhcpv4.DHCPv4,
 
 	return bootfile, nextServer
 }
+
+// encodeClasslessStaticRoutes encodes static routes according to RFC 3442 format.
+// RFC 3442 defines the format for DHCP option 121 (Classless Static Route Option).
+// The format is: width | destination descriptor | router
+// where:
+// - width: number of significant bits in the subnet mask (0-32)
+// - destination descriptor: variable length based on width
+// - router: 4 bytes for IPv4 address of the router
+func encodeClasslessStaticRoutes(routes []data.StaticRoute) ([]byte, error) {
```

Comment:
FYI, the https://github.com/insomniacslk/dhcp/blob/master/dhcpv4/option_routes.go  package provides this same functionality. It has a [`Route`](https://github.com/insomniacslk/dhcp/blob/d3abe7ccb0ad34d401030e9e1365c83075732563/dhcpv4/option_routes.go#L12) struct and a [`ToBytes`](https://github.com/insomniacslk/dhcp/blob/d3abe7ccb0ad34d401030e9e1365c83075732563/dhcpv4/option_routes.go#L78) function.